src/devices/src/virtio/queue.rs

```diff
@@ -309,31 +330,40 @@ impl Queue {
     }
 
     /// Puts an available descriptor head into the used ring for use by the guest.
-    pub fn add_used(&mut self, mem: &GuestMemoryMmap, desc_index: u16, len: u32) {
+    pub fn add_used(
+        &mut self,
+        mem: &GuestMemoryMmap,
+        desc_index: u16,
+        len: u32,
+    ) -> Result<(), QueueError> {
         if desc_index >= self.actual_size() {
             error!(
                 "attempted to add out of bounds descriptor to used ring: {}",
                 desc_index
             );
-            return;
+            return Ok(());
```

Comment:
This should be an error. You can add a special variant for it like `QueueError::DescIndexOutOfBounds(u16)`

Reply:
Done